### PR TITLE
refactor alumni stories and news/events to use a management command t…

### PR DESCRIPTION
**Pull request by richbrennan**
…o fetch from api

Adds a management command and related code to fetch content from the old site api outside of the page request proces.

### CODE REVIEW

File: rca/api_content/content.py [hunk 1]
```diff
@@ -36,32 +41,28 @@ def ranged_date_format(date, date_to):
...
+            f"Error occured when fetching further detail data from {url}"
         )
     except (HTTPError, ConnectionError):
         logger.exception(
-            "HTTP/ConnectionError occured when fetching further detail data from {url}"
+            f"HTTP/ConnectionError occured when fetching further detail data from {url}"
         )
         raise CantPullFromRcaApi(
-            "Error occured when fetching further detail data from {url}"
+            f"Error occured when fetching further detail data from {url}"
```
Comment (kevinhowbrook) [inline rca/api_content/content.py:58]: as if I forgot the f string :facepalm:

File: rca/api_content/content.py [hunk 2]
```diff
@@ -274,3 +275,41 @@ def pull_alumni_stories(programme_type_slug=None):
...
+
+
+class _BaseContentAPI:
+    def __init__(self, func, cache_key):
+        self.func = func
+        self.cache_key = cache_key
+
+    def fetch_from_api(self):
+        try:
+            data = self.func()
```
Comment (kevinhowbrook) [inline rca/api_content/content.py:287]: I've not seen this pattern before but at instantiation wil func = pull_alumni_stories/pull_news_and_events ?

File: rca/api_content/content.py (hunk 1 above)
Comment (richbrennan) [reply rca/api_content/content.py:58]: I missed it in code review 😬

File: rca/api_content/content.py (hunk 2 above)
Comment (richbrennan) [reply rca/api_content/content.py:287]: Yes, functions are first class objects so they can be assigned/passed around etc, then called when needed.

Comment (richbrennan) [reply rca/api_content/content.py:287]: TBH this class inheritance might not be the best pattern, but it separated the code into command and query so they can be used in different contexts.

Passing functions around and calling them dynamically is fine though.

**Pull request merged**